### DISCUSSION TITLE
Манипулирование плейлистами

### DISCUSSION THREAD

**Discussion by deNoi5e** [MarshalX/yandex-music-api#631] [Q&A] [answered]
Добрый день. Бегло побегал по реализованным методам и не нашел сходу работу с плейлистом. Интересуют методы, которые умеют добавлять данные в плейлист, по содержимому .csv-файла. Интересует не реализация хотелки, а умеем ли мы сейчас по api манипулировать плейлистом (создание плейлиста/наполнение треками/удаление плейлиста/удаление трека из плейлиста)? Ткните пожалуйста носом в доку, или, если не реализовано - есть ли возможность реализовать?

**Comment by MarshalX** [comment, marked as answer]
Добрый. Есть методы создания плейлиста, изменения, вставки и удаления в него треков и другие. 

Вот, например, метод добавления трека в плейлист https://yandex-music.readthedocs.io/en/main/client.html#yandex_music.client.Client.users_playlists_insert_track

Кроме этого есть вспомогательный класс с помощью которого можно сделать много операций вставок/удаления в плейлист за один запрос. Вам нужно используя данный класс. Создать его экземпляр и заполнить данными из csv. Предварительно найдя все необходимые треки из файла в базе ЯМ другим запросом (метод .tracks)

Документация к вспомогательному классу: https://yandex-music.readthedocs.io/en/main/yandex_music.utils.difference.html#yandex_music.utils.difference.Difference

**Reply by deNoi5e** [reply]
Спасибо, буду изучать!
